spncreation.ps1

```diff
@@ -41,57 +41,57 @@ function Get-Password
     }
 }
 
-#Initialize
+# Initialize
 $ErrorActionPreference = "Stop"
 $VerbosePreference = "SilentlyContinue"
 $userName = ($env:USERNAME).Replace(' ', '')
 $newguid = [guid]::NewGuid()
-$displayName = [String]::Format("VSTS.{0}.{1}", $userName, $newguid)
+$displayName = [String]::Format("AzDevOps.{0}.{1}", $userName, $newguid)
 $homePage = "http://" + $displayName
 $identifierUri = $homePage
 
 
-#Initialize subscription
-$isAzureModulePresent = Get-Module -Name AzureRM* -ListAvailable
+# Initialize subscription
+$isAzureModulePresent = Get-Module -Name Az -ListAvailable
 if ([String]::IsNullOrEmpty($isAzureModulePresent) -eq $true)
 {
-    Write-Output "Script requires AzureRM modules to be present. Obtain AzureRM from https://github.com/Azure/azure-powershell/releases. Please refer https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/DeployAzureResourceGroup/README.md for recommended AzureRM versions." -Verbose
+    Write-Output "Script requires Azure PowerShell modules to be present. Obtain Azure PowerShell from https://github.com/Azure/azure-powershell/releases." -Verbose
     return
 }
 
-Import-Module -Name AzureRM.Profile
-Write-Output "Provide your credentials to access Azure subscription $subscriptionName" -Verbose
-Login-AzureRmAccount -SubscriptionName $subscriptionName -EnvironmentName $environmentName
-$azureSubscription = Get-AzureRmSubscription -SubscriptionName $subscriptionName
-$connectionName = $azureSubscription.SubscriptionName
+Import-Module -Name Az.Accounts
+Write-Output "Provide your credentials to access your Azure subscription $subscriptionName" -Verbose
+Login-AzAccount -SubscriptionName $subscriptionName -EnvironmentName $environmentName
+$azureSubscription = Get-AzSubscription -SubscriptionName $subscriptionName
+$connectionName = $azureSubscription.Name
 $tenantId = $azureSubscription.TenantId
 $id = $azureSubscription.SubscriptionId
 
 
-#Create a new AD Application
+# Create a new AD Application
 Write-Output "Creating a new Application in AAD (App URI - $identifierUri)" -Verbose
 $password = Get-Password
-$azureAdApplication = New-AzureRmADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
+$azureAdApplication = New-AzADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
```

Comment:
This command will not work with Az PowerShell 7 and above. It is using Microsoft Graph sot the password cannot be set and the identifier Uri must use a verified domain in of the organization or its subdomain.

```suggestion
$azureAdApplication = New-AzADApplication -DisplayName $displayName -HomePage $homePage -Verbose
```